### PR TITLE
Add "push" command

### PR DESCRIPTION
**Pull request by mreid-moz**
Add basic support to push queries back to STMO. This requires a few
small changes to "track" to add the required query metadata
fields.

### CODE REVIEW

**Comment by mreid-moz** [comment]
This is a work in progress, I wanted to get feedback on the approach before tackling test coverage.

I decided to switch to using filenames for the query identifier for tab-completion friendliness.

**Comment by mreid-moz** [comment]
@harterrt RFAL.

File: stmocli/cli.py [hunk 1]
```diff
@@ -23,26 +24,67 @@ def init(conf):
...
+    try:
+        meta = conf.get_query(file_name)
+        with open(file_name, 'r') as fin:
+            query = fin.read()
+
+        try:
+            redash.update_query(meta['id'], meta['name'],
+                                query, meta['data_source_id'],
+                                meta['description'], meta['options'])
+            m = md5.new()
```
Comment (harterrt) [inline stmocli/cli.py]: Why include MD5? Testability?

File: tests/test_cli.py [hunk 2]
```diff
@@ -34,31 +45,105 @@ def test_init(runner):
...
+        m.update(query_after)
+        expected_output = "Query ID {} updated with content from {} (md5 {})".format(
+            query_id, file_name, m.hexdigest())
+        assert push_result.output.strip() == expected_output
+
+def test_push_fail(runner):
+    query_id = '49741'
+    file_name = 'poc.sql'
+
+    with runner.isolated_filesystem():
```
Comment (harterrt) [inline tests/test_cli.py]: Totally optional suggestion: Consider refactoring the duplicated code into a separate function.

File: README.md [hunk 3]
```diff
@@ -70,14 +74,14 @@ Be sure to use version control.
...
-**Not Yet Implemented**
+**Implemented!**
 
-`stmocli push [<id>]`
+`stmocli push [<filename>]`
 
-Pushes the current SQL statements and metadata to re:dash for the given query.
-If no query id is specified, push data for all queries.
+Pushes the current SQL statements and metadata to re:dash for the given query file.
+If no filename is specified, push data for all queries.
```
Comment (harterrt) [inline README.md]: It looks like filename is required in the current implementation.

File: stmocli/cli.py (hunk 1 above)
Comment (mreid-moz) [reply stmocli/cli.py]: Yup, I was having trouble coming up with meaningful tests, this gives an easily double-checkable way to make sure the correct contents were pushed.

File: tests/test_cli.py (hunk 2 above)
Comment (mreid-moz) [reply tests/test_cli.py]: fixed

File: README.md (hunk 3 above)
Comment (mreid-moz) [reply README.md]: I updated the `README` for now, I'll file a follow-up issue for supporting "push all queries"

**Comment by mreid-moz** [comment]
I don't have permission to merge, looks like you'll have to.

**Comment by mreid-moz** [comment]
Oh, and yes I tried this with a [real query](https://sql.telemetry.mozilla.org/queries/51064/source#137010), it worked as expected.

**Comment by harterrt** [comment]
I fixed the repo permissions and I'm merging now. Thanks again.

**Pull request merged**